### PR TITLE
Update fuchsia-tester so it takes map of kernel test files

### PR DESCRIPTION
**Pull request by aam**
This switches fuchsia-tester to run against whole precompiled test suite as running one precompiled test at a time is very slow.

### CODE REVIEW

File: packages/flutter_tools/bin/fuchsia_tester.dart [hunk 1]
```diff
@@ -54,9 +53,8 @@ Future<Null> run(List<String> args) async {
     ..addOption(_kOptionShell, help: 'The Flutter shell binary')
     ..addOption(_kOptionTestDirectory, help: 'Directory containing the tests')
     ..addOption(_kOptionSdkRoot, help: 'Path to the SDK platform files')
-    ..addOption(_kOptionTestFile, help: 'Test file to execute')
-    ..addOption(_kOptionDillFile, help: 'Precompiled dill file for test')
     ..addOption(_kOptionIcudtl, help: 'Path to the ICU data file')
+    ..addOption(_kOptionPrecompiledDillFiles, help: 'Path to json file that maps dart test file ot precompiled dill')
```
Comment (tvolkert) [inline packages/flutter_tools/bin/fuchsia_tester.dart]: "... that maps Dart test files to precompiled dill files"

File: packages/flutter_tools/bin/fuchsia_tester.dart [hunk 2]
```diff
@@ -27,17 +28,15 @@ const String _kOptionPackages = 'packages';
...
 const List<String> _kRequiredOptions = <String>[
   _kOptionPackages,
   _kOptionShell,
   _kOptionTestDirectory,
   _kOptionSdkRoot,
-  _kOptionTestFile,
-  _kOptionDillFile,
-  _kOptionIcudtl
+  _kOptionIcudtl,
+  _kOptionPrecompiledDillFiles
```
Comment (tvolkert) [inline packages/flutter_tools/bin/fuchsia_tester.dart]: Add a trailing comma

File: packages/flutter_tools/bin/fuchsia_tester.dart [hunk 3]
```diff
@@ -27,17 +28,15 @@ const String _kOptionPackages = 'packages';
 const String _kOptionShell = 'shell';
 const String _kOptionTestDirectory = 'test-directory';
 const String _kOptionSdkRoot = 'sdk-root';
-const String _kOptionTestFile = 'test-file';
-const String _kOptionDillFile = 'dill-file';
 const String _kOptionIcudtl = 'icudtl';
+const String _kOptionPrecompiledDillFiles = 'precompiled-list';
```
Comment (tvolkert) [inline packages/flutter_tools/bin/fuchsia_tester.dart]: Maybe just call this "tests"?

File: packages/flutter_tools/bin/fuchsia_tester.dart [hunk 4]
```diff
@@ -115,13 +111,22 @@ Future<Null> run(List<String> args) async {
       collector = new CoverageCollector();
     }
 
+
+    Map<String, String> precompiledDillFiles = {};
+    List<Map<String, dynamic>> json = List<Map<String, dynamic>>.from(
+      jsonDecode(
+        fs.file(argResults[_kOptionPrecompiledDillFiles]).readAsStringSync()));
+    for (Map<String, dynamic> map in json) {
+      precompiledDillFiles[map["source"]] = map["dill"];
```
Comment (tvolkert) [inline packages/flutter_tools/bin/fuchsia_tester.dart]: Why not just have the raw JSON content be the desired map?

```
{
  "foo": "bar",
  "baz": "qux"
}
```

File: packages/flutter_tools/bin/fuchsia_tester.dart [hunk 5]
```diff
@@ -115,13 +111,22 @@ Future<Null> run(List<String> args) async {
       collector = new CoverageCollector();
     }
 
+
+    Map<String, String> precompiledDillFiles = {};
+    List<Map<String, dynamic>> json = List<Map<String, dynamic>>.from(
+      jsonDecode(
```
Comment (tvolkert) [inline packages/flutter_tools/bin/fuchsia_tester.dart]: Nit: elsewhere in flutter tools, we use `json.decode()`

File: packages/flutter_tools/lib/src/test/flutter_platform.dart [hunk 6]
```diff
@@ -449,7 +453,7 @@ class _FlutterPlatform extends PlatformPlugin {
           ? precompiledDillPath
```
Comment (tvolkert) [inline packages/flutter_tools/lib/src/test/flutter_platform.dart]: This should be referencing the right dill path for the test in question.

**Review by tvolkert — COMMENTED** [review]
Bots are complaining...

File: packages/flutter_tools/bin/fuchsia_tester.dart (hunk 4 above)
Comment (aam) [reply packages/flutter_tools/bin/fuchsia_tester.dart]: That's what I started with, but then had difficulty generating such json using gn write_file()

**Review by tvolkert — APPROVED** [review]
LGTM

**Pull request merged**